src/Relation/Nullary/Decidable.agda

```diff
@@ -52,8 +51,8 @@ via-injection inj _≟_ x y = map′ injective cong (to x ≟ to y)
 -- A lemma relating True and Dec
 
 True-↔ : (a? : Dec A) → Irrelevant A → True a? ↔ A
-True-↔ (true  because [a]) irr = mk↔ₛ′ (λ _ → invert [a]) _ (irr (invert [a])) cong′
-True-↔ (false because ofⁿ ¬a) _ = mk↔ₛ′ (λ ()) (invert (ofⁿ ¬a)) (⊥-elim ∘ ¬a) λ ()
+True-↔ (true  because [a]) irr = let a = invert [a] in mk↔ₛ′ (λ _ → a) _ (irr a) cong′
+True-↔ (false because [¬a]) _  = let ¬a = invert [¬a] in mk↔ₛ′ (λ ()) ¬a (λ a → contradiction a ¬a) λ ()
```

Comment:
The implicits break putting just `(flip contradiction ¬a) here:
```agda
/home/james/RESEARCH/AGDA/Agda-stdlib-dev-v2.1/src/Relation/Nullary/Decidable.agda:55,75-96
Cannot instantiate the metavariable _C_128 to solution
(λ ()) (invert [¬a] x) ≡ x since it contains the variable x
which is not in scope of the metavariable
when checking that the expression flip contradiction ¬a has type
Function.Definitions.StrictlyInverseˡ _≡_ (λ ()) ¬a
```
and I'm stubborn enough to want to not see `⊥-elim` *anywhere* outside `Relation.Nullary.Negation.Core` or `Data.Empty`... (part of the point of this PR)